driver/src/test/functional/com/mongodb/client/CommandMonitoringTest.java

```diff
@@ -314,6 +314,8 @@ private void executeOperation() {
                     BsonDocument operation = definition.getDocument("operation");
                     if (operation.containsKey("read_preference")) {
                         commandDocument.put("$readPreference", operation.getDocument("read_preference"));
+                    } else if (!isDiscoverableReplicaSet() && !isSharded()) {
```

Comment:
The need for this is somewhat controversial.  Currently debating it with Jesse and Mathias.  The spec seems to require it, but Mathias argues that, at least for writes, it doesn't make sense to added primaryPreferred read preference

Reply:
Ack